### ISSUE TITLE
Cannot open Single Player

### ISSUE THREAD

**Issue by AstoriaWinters** [lucaargolo/extra-generators#4]
Hello. When I try to open a single-player world with this mod in it boots me to the Multiplayer cannot connect screen. Taking this mod out seems to fix that- the specific mods I find it does this with are: Lifts and Happiness is a Warm Gun. I am unsure of what the issue may be, unfortunately. It also does not generate a crash report.

**Comment by Chuunkyz** [comment]
same here

**Comment by lucaargolo** [comment]
Could you please send the latest.log file?

**Comment by AstoriaWinters** [comment]
Here is a log file, I just created it, so it has the full list of mods I am currently using on my server. But they all work 100% on the server. it's just trying to load single player it fails lol
[latest.log](https://github.com/lucaargolo/extra-generators/files/7145599/latest.log)

**Comment by lucaargolo** [comment]
Looks like a problem with NbtCrafting. Could you try removing the mod and seeing if it fixes it?

**Comment by AstoriaWinters** [comment]
So, I had no idea what mod that was a part of so went through again to try and figure it out. I had updated a few mods so the list I had for conflicts was smaller now, it seems to be I guess nbt data mods, trade and polymorph seem to be the issues. they both change nbt data some I guess? Is there a way to make it so they do not cause issues maybe, please?
[latest.log](https://github.com/lucaargolo/extra-generators/files/7145835/latest.log) (the log only has the issue mods in it atm)

**Comment by lucaargolo** [comment]
This log has nothing to do with this mod... Something else broke.

**Comment by AstoriaWinters** [comment]
it works fine without the generators mod in the mod list though, so I'm not sure how it could not have anything to do with the problem?

**Comment by lucaargolo** [comment]
Well, the first one you sent was being caused by NbtCrafting. The second one successfully loaded everything from the mod and then failed at something else.
![image](https://user-images.githubusercontent.com/49497195/132907939-ba913a84-a5f6-4408-86a3-d6e114529944.png)

**Comment by AstoriaWinters** [comment]
I have no idea really. The whole modlsit works 100% fine without the generators, and I do not have a "NbtCrafting" mod installed that I can find. I've went thru my list a few times now to try and find the issue and cannot, I jsut know it only shows up with this mod installed. 

I found the mod that caused the issue from the 2nd log, removed it. game loaded fine, closed minecraft and tried loading the world again and i get the same error.

I jsut tested it again, with the generator mod, I get the error. If I remove the generator mod, I can log into the game fine. here are the 2 logs of that. The fact that the game can load the same save jsut fine without the generators tells me its an issue with them, I guess interactign with something? But what that is I have no idea since I do no have a "NbtCrafting" mod.

[with generators log.log](https://github.com/lucaargolo/extra-generators/files/7145988/with.generators.log.log)
[without generators log.log](https://github.com/lucaargolo/extra-generators/files/7145989/without.generators.log.log)

**Comment by lucaargolo** [comment]
Well, your logs are very cluttered but here you can see that everything loaded fine in your latest attempt with the generators mod.
![image](https://user-images.githubusercontent.com/49497195/132911970-0ec09093-4c9d-46c0-9acd-000de4f376bb.png)
Then you got kicked for invalid player data. Something, not related to my mod at all. 
![image](https://user-images.githubusercontent.com/49497195/132912204-903a498f-3f89-4507-a2d2-04148dcceee3.png)
Basically, every log you sent had a different error.

**Comment by AstoriaWinters** [comment]
Ok not sure then. It loads jsut fine without the generators as I have said. Literally the only difference in those 2 last logs in my mod list was the generators mod. I'll jsut continue not using them in single player I guess then. Thank you for trying to help though

**Comment by lucaargolo** [comment]
Well, in the first log the problem was probably happening because of the NBTCrafting mod. Maybe something got corrupted afterward? I don't know. Could you try creating a new world and seeing if it works? Maybe that can help us to solve your problem.

**Comment by AstoriaWinters** [comment]
So, made a brand new world. 1st load- everything works fine. 2nd load after closing mc down- "invalid player data". 3rd load- remove the generators mod. Everything works fine.

1st load, everything is fine. 
[1st load.log](https://github.com/lucaargolo/extra-generators/files/7146379/1st.load.log)

Close and reopen the game, Cannot load into the same world.  "invalid player data" 
[2nd load.log](https://github.com/lucaargolo/extra-generators/files/7146392/2nd.load.log)

Remove the generator mod and load into the world. Works fine.  
[3rd load.log](https://github.com/lucaargolo/extra-generators/files/7146415/3rd.load.log)

**Comment by AstoriaWinters** [comment]
this is what is confusing me the most- because it works on the 1st load. but something seems to go wrong between opening the game a 2nd time that makes it glitch out, and if I remove the generators mod it then works fine.

**Comment by lucaargolo** [comment]
Alright, so after searching for a while I stumbled upon [this post](https://www.reddit.com/r/Minecraft/comments/odnj71/singleplayer_connection_lost_invalid_player_data/) on Reddit where someone is having a problem similar to yours, but this person does not have the Extra Generators mod, instead he managed to fix it by removing the Iris shaders mod. My theory is that another mod in your modpack is breaking something (probably the same mod that was causing issues in the reddit post), but it's only getting triggered when you have the extra generators mod (or the iris shader mod) installed as well. 
I strongly recommend you to try and find what other mod is breaking Extra Generators, because while removing EG will solve your problem, for now, it may appear later on in your gameplay and even corrupt your world.

**Comment by AstoriaWinters** [comment]
So, I have my mods all seperated by "Required" "Optional" and "Optifine Replacement(which is Iris and such)" the last batch of logs i sent is only the "Required" mods. Which is only the mods mc tells you you need to join my current Test server, So I do not have my Iris or anything else installed at the time. I have tried to batch add mods to try and find the issue and I always seem to come up with a different batch of mods that causes it, but I can never replicate it with those mods again. The only consistant mod that is part of the problem is always Generators. I'll try and do some more testing after I sleep some though and see if I maybe missed something.

Thank you for your time!

**Comment by DEWSTREAMING** [comment]
Hey I'm having the same issue of not joining single player, getting booted to the multiplayer screen with connection lost, but the mods specifically are Immersive Portals, and Immersive Portals Survival Adaptation and obviously Extra Generators
[latest.log](https://github.com/lucaargolo/extra-generators/files/7340453/latest.log)
 that's my latest log, not sure if I should have made a new issue since it's with different mods, but it seems to be the same problem, might help narrow down the issue? Disabling either the Extra Generators, or the Immersive Portals Survival Adaptation allows me to join single player just fine, this is the same even with over 40 other mods. These 2 specifically don't want to work together.
 
 *EDIT*
 To clarify this was on a completely knew instance and with a completely new world being generated
